build.gradle

```diff
@@ -37,59 +29,52 @@ repositories {
     mavenCentral()
 }
 
-configurations.compile {
+configurations.implementation {
     exclude group: 'org.springframework.boot', module: 'spring-boot-starter-tomcat'
 }
 
 dependencies {
-    compile group: 'javax.inject', name: 'javax.inject', version: '1'
-
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter'
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-mail'
-
-    compile group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310'
-    compile group: 'com.squareup.okhttp3', name:'okhttp', version:'4.9.1'
-
-    compile group: 'org.knowm.xchange', name: 'xchange-quoine', version: '5.0.8'
-    compile group: 'org.knowm.xchange', name: 'xchange-stream-kraken', version: '5.0.8'
-    compile group: 'org.knowm.xchange', name: 'xchange-bitflyer', version: '5.0.8'
-    compile group: 'org.knowm.xchange', name: 'xchange-stream-coinbasepro', version: '5.0.8'
-    compile group: 'org.knowm.xchange', name: 'xchange-cexio', version: '5.0.8'
-    compile group: 'org.knowm.xchange', name: 'xchange-poloniex', version: '5.0.8'
-    compile group: 'org.knowm.xchange', name: 'xchange-stream-gemini', version: '5.0.8'
-    compile group: 'org.knowm.xchange', name: 'xchange-bitstamp', version: '5.0.8'
-    compile group: 'org.knowm.xchange', name: 'xchange-cobinhood', version: '5.0.8'
-    compile group: 'org.knowm.xchange', name: 'xchange-binance', version: '5.0.8'
-
-    compile group: 'com.github.seratch', name: 'jslack', version: '3.4.2'
-    compile group: 'javax.websocket', name: 'javax.websocket-api', version: '1.1'
-
-    compile group: 'org.apache.commons', name: 'commons-collections4', version: '4.4'
-    compile group: 'org.apache.commons', name: 'commons-text', version: '1.9'
-    compile group: 'commons-io', name: 'commons-io', version: '2.9.0'
-
-    testCompile group: 'junit', name: 'junit', version: '4.13.2'
-    testCompile group: 'org.mockito', name: 'mockito-core', version: '3.10.0'
+    implementation group: 'javax.inject', name: 'javax.inject', version: '1'
```

Comment:
Why do we need javax.inject? I think we only use spring dependency injection. I know this has been in the build file for quite some time but now that I see it I decided to ask. Maybe we can remove it?

Reply:
Honestly I just add it to any Spring project out of habit now, but if I remember correctly without it you can use the `@Autowired` annotation but not `@Inject`. If you're using Spring they both do exactly the same thing. I prefer to stick to the platform agnostic annotations when I can even though I know that realistically we're never switching from Spring to anything else in this project. Just personal preference really.